### PR TITLE
[#35961 M4] Add legalese step without any checks

### PR DESCRIPTION
**Pull request by oisincoveney**
Epic #35961 - Milestone 4

### Description

Adds the `Legalese` step to the static embedding modal flow for OSS users. Right now, there are no checks or actions done to indicate whether or not the user has actually accepted the terms - clicking "Agree" simply moves the user to the static embedding settings. This extra behavior (which will most likely require some BE work) will be sorted in #37327 .

### How to verify

* Ensure embedding is enabled. 
* Go to a question/dashboard and click the share dropdown menu, and click `Embed`.
* Click `Static embed`.
* There should be a legalese step. You should see a link to the embedding terms, and clicking `Agree` should take you to the static embedding settings for that question/dashboard

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Comment by oisincoveney** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/metabase/metabase/37008?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `master`
  * **PR #37006** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37006?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #37008** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37008?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/metabase/metabase/37008?utm_source=stack-comment).

File: frontend/src/metabase/public/components/widgets/EmbedModal.tsx
Comment (oisincoveney) [inline frontend/src/metabase/public/components/widgets/EmbedModal.tsx:54]: This function will be expanded to check if the user has already accepted the legalese in the future. Right now it's just cycling through the states

File: frontend/src/metabase/public/components/widgets/EmbedModalContent.jsx
Comment (oisincoveney) [inline frontend/src/metabase/public/components/widgets/EmbedModalContent.jsx:115]: The ternary is getting a little much here, so split it out into `if` statements

File: frontend/src/metabase/public/components/widgets/LegaleseStep/LegaleseStep.tsx
Comment (oisincoveney) [inline frontend/src/metabase/public/components/widgets/LegaleseStep/LegaleseStep.tsx:17]: Will splitting this sentence mess up the translation? I only ask because splitting the string here, between `…to` and `our embedding license` might introduce some issues when translating to languages that depend on prepositions for cases. I might be misunderstanding how the translation works here but I've been getting uneasy about splitting strings up like this so just want to make sure.

The issue exists in Estonian but I'll put a Russian example since @deniskaber can more easily verify it better than I can. I also know it's google translate which might be very wrong, but I think it still can get the general idea across

image

image

(i.e. `нашей лицензией` vs `наша лицензия`)

File: frontend/src/metabase/public/components/EmbedModal/EmbedModal.tsx
Comment (oisincoveney) [inline frontend/src/metabase/public/components/EmbedModal/EmbedModal.tsx:48]: I'd much rather have these `goTo` functions as explicitly `goToNext` and `goToPrevious` since we're going to need to add some logic to the `goToNext` function, while `goToPrevious` will always return us to the landing. I didn't want to reuse `setEmbedType` since someone else might look at this code and have no idea what `setEmbedType("application")` or `setEmbedType("legalese")` means at first glance

File: frontend/src/metabase/public/components/EmbedModal/EmbedModal.unit.spec.tsx
Comment (oisincoveney) [inline frontend/src/metabase/public/components/EmbedModal/EmbedModal.unit.spec.tsx:14]: let me know if there's a better way to handle this, I'm not exactly a fan

File: frontend/src/metabase/public/components/widgets/LegaleseStep/LegaleseStep.styled.tsx
Comment (npretto) [inline frontend/src/metabase/public/components/widgets/LegaleseStep/LegaleseStep.styled.tsx:9]:
```suggestion
  border: 1px solid ${({ theme }) => theme.colors.border[0]};
  border-radius: ${({ theme }) => theme.radius.md};
```

I haven't found other places where we do `${({ theme }) => css`

(If you accept this, we'll need to remove the css import too)

Comment (oisincoveney) [reply frontend/src/metabase/public/components/widgets/LegaleseStep/LegaleseStep.styled.tsx:9]: Hmm, I'm not sure I like this change. What's the benefit over the original? I like my solution since it keeps the number of arrow functions to a minimum, but if you have a better reason I'd be happy to change it

Comment (npretto) [reply frontend/src/metabase/public/components/widgets/LegaleseStep/LegaleseStep.styled.tsx:9]: It's mostly for consistency with how we usually do it, I checked emotion's docs and I haven't find any example of using `css` inside a styled like that.
The docs mostly use tagged literals as in my suggestions with inline interpolation, or use the object syntax like this:
```tsx
const Container = styled.div(props => ({
  display: 'flex',
  flexDirection: props.column && 'column'
}))
```

Anyway, this is a nit/personal opinion so feel free to ignore it

Comment (oisincoveney) [reply frontend/src/metabase/public/components/widgets/LegaleseStep/LegaleseStep.styled.tsx:9]: Makes sense. I have a bias towards consistency over what makes me feel good so I'll change it 🙂

File: frontend/src/metabase/public/components/widgets/LegaleseStep/LegaleseStep.tsx
Comment (npretto) [reply frontend/src/metabase/public/components/widgets/LegaleseStep/LegaleseStep.tsx:17]: Aren't our translations manual/hand-written anyway?
As far as I understand it, the translator will just have 2 keys/"original sentences" to translate but they can be translated as we want.

I looked if using `jt` helps with that, I believe I found the issue where jt originated from: https://github.com/ttag-org/ttag/issues/3#issuecomment-285283686 (funny that the usecase case is also translating something with a link in it :) )
If what they wrote it's correct it seems like even with jt the full context will be missed (as it creates two "keys").

Comment (oisincoveney) [reply frontend/src/metabase/public/components/widgets/LegaleseStep/LegaleseStep.tsx:17]: Ok then let's leave it as it is, and let the translators handle it

File: frontend/src/metabase/public/components/EmbedModal/EmbedModal.unit.spec.tsx
Comment (WiNloSt) [reply frontend/src/metabase/public/components/EmbedModal/EmbedModal.unit.spec.tsx:14]: I'd not set `full: true` because in [milestone 5 ](https://github.com/metabase/metabase/pull/36181), the modal would not be full anyway. This prevents `Modal` from rendering `RoutelessFullPageModal` which causes the error you mentioned.

This might be worth the compromise, but let's discuss with Product too.

You need to play around with the modal size a bit. In this example I pass `wide: true` and `fit: embedType !== "application"` to `Modal` component in `EmbedModal`

image

Otherwise, if we decide to keep this mock, let's add a comment to ensure we won't forget to revisit this after milestone 5 since we won't render `RoutelessFullPageModal` in embed modal anymore.

File: frontend/src/metabase/public/components/widgets/EmbedModalContent.jsx
Comment (WiNloSt) [reply frontend/src/metabase/public/components/widgets/EmbedModalContent.jsx:115]: This looks a lot easier to understand 👍

File: frontend/src/metabase/public/components/widgets/LegaleseStep/LegaleseStep.styled.tsx
Comment (WiNloSt) [reply frontend/src/metabase/public/components/widgets/LegaleseStep/LegaleseStep.styled.tsx:9]: Both look fine IMO. But if you have more than a handful of properties that need `theme`, I'd be biased toward @oisincoveney 's solution more because we don't have to repeat `theme` every time.

File: frontend/src/metabase/public/components/widgets/LegaleseStep/LegaleseStep.tsx
Comment (WiNloSt) [reply frontend/src/metabase/public/components/widgets/LegaleseStep/LegaleseStep.tsx:17]: The difference would be what would be presented to the translators.

In this case, there would be 2 strings
1. "By clicking “Agree and continue” you're agreeing to"
2. "our embedding license."

if you use `jt` e.g. 
```js
jt`By click... ${(<ExternalLink ...>{t`our embedding license.`}</ExternalLink>)}`
```

the translators would see
1. "By clicking “Agree and continue” you're agreeing to {0}"
2. "our embedding license."

which might give them a bit more context that there is some more texts after the first sentence.

tl;dr I like the use of `jt` in this case.

**Review by WiNloSt — APPROVED** [review]
Looks good to me, but I'd wait for @deniskaber 's review before merging since his milestone 5 PR has also modified many files here.

**Review by deniskaber — APPROVED** [review]

File: frontend/src/metabase/public/components/EmbedModal/EmbedModal.tsx
Comment (oisincoveney) [reply frontend/src/metabase/public/components/EmbedModal/EmbedModal.tsx:48]: Yeah I think that's a good call. I'll change it to `goBackToEmbedModal` or something like that

File: frontend/src/metabase/public/components/widgets/LegaleseStep/LegaleseStep.tsx
Comment (oisincoveney) [reply frontend/src/metabase/public/components/widgets/LegaleseStep/LegaleseStep.tsx:17]: Ok I think the `jt` will work just fine here - it's weird that it still gives two keys (I mean, I understand why, but still) but if we can give the translators as much context then let's give it to them

File: frontend/src/metabase/public/components/widgets/LegaleseStep/LegaleseStep.styled.tsx
Comment (oisincoveney) [reply frontend/src/metabase/public/components/widgets/LegaleseStep/LegaleseStep.styled.tsx:9]: Ok, I'll leave it as is for now then

File: frontend/src/metabase/public/components/EmbedModal/EmbedModal.unit.spec.tsx
Comment (oisincoveney) [reply frontend/src/metabase/public/components/EmbedModal/EmbedModal.unit.spec.tsx:14]: Yeah, I was thinking about that as well. My thinking was that I should try and reduce the amount of user-facing changes in this milestone since I know Denis is going to move things around, so I wanted to keep things relatively predictable for him

Comment (oisincoveney) [reply frontend/src/metabase/public/components/EmbedModal/EmbedModal.unit.spec.tsx:14]: I like this huge modal style for the time-being so we can reduce the code size, since Milestone 5 is coming very soon 😄

File: frontend/src/metabase/dashboard/containers/DashboardSharingEmbeddingModal/DashboardSharingEmbeddingModal.tsx
Comment (oisincoveney) [reply frontend/src/metabase/dashboard/containers/DashboardSharingEmbeddingModal/DashboardSharingEmbeddingModal.tsx:5]: fixed 😄

File: frontend/src/metabase/public/components/EmbedModal/EmbedModal.unit.spec.tsx
Comment (oisincoveney) [reply frontend/src/metabase/public/components/EmbedModal/EmbedModal.unit.spec.tsx:14]: Nice! So I used your suggestion and I could immediately remove all of the mock code and it still looks good 😄

File: frontend/src/metabase/public/components/EmbedModal/EmbedModal.tsx
Comment (oisincoveney) [reply frontend/src/metabase/public/components/EmbedModal/EmbedModal.tsx]: Done!

**Pull request merged**
